pkg/controller/envoy/ads_event.go

```diff
@@ -162,20 +162,20 @@ func (svc *ServiceEvent) handleCdsResponse(rsp *service_discovery_v3.DiscoveryRe
 
 	// when the list of eds typed clusters subscribed changed, we should resubscrbe to new eds.
 	if !slices.EqualUnordered(svc.DynamicLoader.edsClusterNames, lastEdsClusterNames) {
-		svc.rqt = newAdsRequest(resource_v3.EndpointType, svc.DynamicLoader.edsClusterNames)
+		svc.rqt = newAdsRequest(resource_v3.EndpointType, svc.DynamicLoader.edsClusterNames, resp.Nonce)
```

Comment:
This is not right. 

We should pass the eds nonce rather than the cds nonce. If this is the first req for eds, we can pass empty